### PR TITLE
Add inflation stop e2e tests and upgrade handler

### PR DESCRIPTION
**Pull request by drklee3**
❗ To be merged after:
- [x] #1766 is merged into master
- [x] `release/v0.25.x` up to date with master

## Description
- Add e2e test for expected module parameters and inflation for:
  - before upgrade
  - after upgrade but before switchover
  - after switchover
- Tests community fund consolidation
- Add upgrade handlers for module state to work properly
  - Includes 3 - mainnet, testnet, and e2e test

### CODE REVIEW

File: tests/e2e/e2e_upgrade_inflation_test.go [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+		suite.Require().NoError(err)
+
+		// Events are not emitted if amount is 0
+		stakingRewardEvents := util.FilterEventsByType(block.BeginBlockEvents, communitytypes.EventTypeStakingRewardsPaid)
+		suite.Require().Empty(stakingRewardEvents, "staking reward events should not be emitted")
+	})
+
+	suite.Run("staking rewards pay out from x/community", func() {
+		// after switchover
+		queryHeight := switchoverHeight
```
[the remark this answers is not in the document]
Comment (drklee3) [reply tests/e2e/e2e_upgrade_inflation_test.go]: yup that sounds good

File: tests/e2e/e2e_upgrade_inflation_test.go [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+
+				found = true
+			}
+		}
+
+		suite.True(found, "staking reward amount should be found in events")
+	})
+
+	// Staking rewards can still be claimed
+	suite.Run("staking rewards claimable after switchover", func() {
```
[the remark this answers is not in the document]
Comment (drklee3) [reply tests/e2e/e2e_upgrade_inflation_test.go:344]: hmm for now added balance check against the rewards query before claiming

File: x/community/types/events.go [hunk 3]
```diff
@@ -6,6 +6,7 @@ const (
 	EventTypeStakingRewardsPaid = "staking_rewards_paid"
 
 	AttributeKeyStakingRewardAmount = "staking_reward_amount"
+	AttributeKeyDisableTime         = "disable_time"
```
Comment (galxy25) [inline x/community/types/events.go:9]: Nit: disable_inflation_time

File: app/upgrades.go [hunk 4]
```diff
@@ -1,3 +1,88 @@
...
+		app.Logger().Info(fmt.Sprintf("running %s upgrade handler", name))
+
+		toVM, err := app.mm.RunMigrations(ctx, app.configurator, fromVM)
+		if err != nil {
+			return toVM, err
+		}
+
+		app.Logger().Info("initializing x/community params")
+		app.communityKeeper.SetParams(ctx, communityParams)
+
```
Comment (galxy25) [inline app/upgrades.go:85]: Log what community params were set to?

**Review by galxy25 — APPROVED** [review]

File: app/upgrades.go (hunk 4 above)
Comment (drklee3) [reply app/upgrades.go:90]: good idea

File: tests/e2e/.env [hunk 5]
```diff
@@ -11,22 +11,22 @@ E2E_RUN_KVTOOL_NETWORKS=true
 E2E_KVTOOL_KAVA_CONFIG_TEMPLATE="master"
 
 # E2E_INCLUDE_IBC_TESTS when true will start a 2nd chain & open an IBC channel. It will enable all IBC tests.
-E2E_INCLUDE_IBC_TESTS=true
+E2E_INCLUDE_IBC_TESTS=false
```
Comment (pirtleshell) [inline tests/e2e/.env:14]: why no ibc tests?

Comment (drklee3) [reply tests/e2e/.env]: Re-enabled 😃 Might have disabled to make the upgrade height lower to be faster

**Review by rhuairahrighairidh — APPROVED** [review]

**Pull request merged**